pkg/cloudprovider/gcp/object_store.go

```diff
@@ -35,10 +35,25 @@ import (
 
 const credentialsEnvVar = "GOOGLE_APPLICATION_CREDENTIALS"
 
+// Interface retriever wraps the GCP SDK functions for accessing object store so they can be faked for testing.
+type retriever interface {
+	// function getWriteCloser returns a struct confirming to the io.WriteCloser interface
```

Comment:
getWriteCloser returns an io.WriteCloser that can be used to upload data to the specified bucket for the specified key.